pkg/networking/cilium/network_policy.yaml

```diff
@@ -0,0 +1,242 @@
+{{- if .managementCluster }}
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-all-kube-system
+  namespace: kube-system
+spec:
+  podSelector: {}
+  ingress:
+  - {}
+  egress:
+  - {}
+  policyTypes:
+  - Ingress
+  - Egress
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: eksa-system
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-all-eksa-system
+  namespace: eksa-system
+spec:
+  podSelector: {}
+  ingress:
+  - {}
+  egress:
+  - {}
+  policyTypes:
+  - Ingress
+  - Egress
+---
+{{- if .gitopsEnabled }}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: flux-system
+---
+apiVersion: networking.k8s.io/v1
+kind: NetworkPolicy
+metadata:
+  name: allow-all-flux-system
+  namespace: flux-system
+spec:
+  podSelector: {}
+  ingress:
+  - {}
+  egress:
+  - {}
+  policyTypes:
+  - Ingress
+  - Egress
+---
+{{- end }}
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: capi-system
```

Comment:
generating the namespaces here because they haven't been created yet. Cilium gets installed first on the workload cluster, followed by CAPI and then eks-a/flux controllers on the mgmt cluster.
The other option would have been to modify clusterctl InitInfrastructure, in every component attach a NetworkPolicy, because otherwise the pods won't come up when using "always" mode. This seems cleaner, and does not affect the cluster